components/tools/OmeroPy/src/omero/gateway/__init__.py

```diff
@@ -952,6 +996,31 @@ def getAnnotation(self, ns=None):
             return AnnotationWrapper._wrap(self._conn, rv[0].child, link=rv[0])
         return None
 
+    def getAnnotationCounts(self, ns=None):
+        """
+        Get the annotion counts for the current object
+        """
+
+        params = omero.sys.ParametersI()
+        params.addId(long(self._oid))
+
+        q = """
+            select al from %sAnnotationLink al
+            left outer join fetch al.child as an
+            where al.parent.id = :id
+            """ % self.OMERO_CLASS
+
+        if ns:
```

Comment:
We don't need to check if ns is not null since an empty string is not null.
When I add a Tag with ```ns = rstring("")```
Using ```if ns is not None:``` I get
```
>>> project.getAnnotationCounts(ns="")
{'CommentAnnotation': 0,
 'FileAnnotation': 0,
 'LongAnnotation': 0,
 'MapAnnotation': 0,
 'OtherAnnotation': 0,
 'TagAnnotation': 1}
```
But without that fix I get
```
>>> project.getAnnotationCounts(ns="")
{'CommentAnnotation': 7,
 'FileAnnotation': 6,
 'LongAnnotation': 4,
 'MapAnnotation': 4,
 'OtherAnnotation': 4,
 'TagAnnotation': 8}
```